### ISSUE TITLE
Some Pushbullet channels not updating

### ISSUE THREAD

**Issue by patrickdrd** [illogical-robot/apkmirror-public#128]
no updates posted at all on pushbullet!

**Comment by archon810** [comment]
Hmm, you're right. https://www.apkmirror.com/apk/mk-dev-llc/tv-show-favs/tv-show-favs-4-5-0-release/

https://www.pushbullet.com/channel-popup?tag=am-1303042955

![image](https://user-images.githubusercontent.com/469543/80830319-9be7af80-8b9d-11ea-8361-63afa0962ec6.png)

I've pinged Pushbullet devs about this because I don't think it's something we can fix ourselves.

**Comment by archon810** [comment]
Telegram is stuck on an older v5 update. https://www.apkmirror.com/apk/telegram-fz-llc/telegram/telegram-6-1-1-release/

https://www.pushbullet.com/channel-popup?tag=am-29045543

![image](https://user-images.githubusercontent.com/469543/80830461-dbae9700-8b9d-11ea-9c00-d7f03718989b.png)

**Comment by archon810** [comment]
https://twitter.com/battierpeeler/status/1323057863556620289

https://www.apkmirror.com/apk/google-inc/chromecast-built-in/

https://www.pushbullet.com/channel-popup?tag=am-28773310
etc.

**Comment by sammyke007** [comment]
This is still bugged, correct?

**Comment by archon810** [comment]
Yeah, I believe this continues to be an issue with Pushbullet. I'm not sure if Pushbullet is actively developed anymore, it's more on life support.

**Comment by archon810** [comment]
Another related issue was reported [here](https://www.apkmirror.com/apk/lidl-digital-international-gmbh-co-kg/lidl-home/lidl-home-1-0-25-release/lidl-home-1-0-25-android-apk-download/#comment-6013037269):

> Dear @ApkMirror, your Pushbullet link for "Lidl Home App Updates" ISN'T working, sadly. 
> 
> Can it be fixed? ( it's - https://www.pushbullet.com/channel-popup?tag=am1177583991 - that channel that cannot be followed. )
> 
> Thanks! Just reporting a bug.

Based on our analysis, the issue seems to be on PB's side.

**Comment by guzba** [comment]
Hello. PB Dev here. Happy to share my thoughts on this.

A handful of years ago now, APK Mirror and PB set up an RSS system to create channels and send pushes out automatically based on RSS feeds. This was a very clever way to get things up and running quickly, which was great, worked out very well. It has not proved to be a good long-term system though.

It has now been several years and debugging RSS issues for this custom made plugin has become a frustrating burden without any benefit to me and not something I can justify spending time on.

The true solution here is fortunately very straightforward. APK Mirror should use our public API to create channels and send pushes to those channels whenever they want. This will be vastly more reliable, simpler, and better in basically every way. It will require APK Mirror to switch away from our custom RSS integration thing I built for them though.

At this point it's basically a choice between having APK Mirror use our public API just the same way as everyone else does, or continue using the RSS integration and accept it is not perfect (or we can remove it altogether if it isn't worth having any longer).

To create channels programmatically: https://docs.pushbullet.com/#create-channel
To send pushes programmatically: https://docs.pushbullet.com/#create-push

Those 2 public API calls are all that will be needed to swap over to a much better system.

**Comment by sammyke007** [comment]
Tnx for clearing this out!

**Comment by archon810** [comment]
@guzba Great to hear from you. My biggest concern with switching to another system is all the existing subscribers to existing channels. Would we be able to migrate these subscriptions to newly created with the public API channels? We need a transition plan because otherwise we would lose thousands of subscribers as you will undoubtedly shut down the feed system once we switch.

**Comment by sammyke007** [comment]
But... I guess there are more none working channels than working channels. Isn't it?

**Comment by archon810** [comment]
I'm not so sure that's true. A lot of channels were working when I spot checked them.

**Comment by guzba** [comment]
@archon810 Existing channels are owned by the APK Mirror PB account IIRC, which means APKM has API control over those channels. I would suggest just using those existing channels but sending pushes to them manually at the correct time.

You could use the API to check if a channel already exists for that APK, and if it does, just stick to the existing channel. If it does not exist, then you'd want to create a new one of course.

This does have a migration question but I really don't think its impossible and am happy to help inform the path of least resistance at least as I see it with my limited insight to APKM internals ofc.

**Comment by archon810** [comment]
Thanks for the additional thoughts. We'll switch to our Telegram group to discuss and strategize with you further when we're ready to tackle this.

**Comment by archon810** [comment]
https://www.apkmirror.com/apk/getverify-lda/getcontact/getcontact-5-9-3-release/ 
Not working:
![image](https://user-images.githubusercontent.com/469543/222844204-6e90b80d-2157-4926-97cc-29ec5d25e77a.png)

**Comment by guzba** [comment]
I have disabled new channel creation because Superfeedr cannot register anymore: https://github.com/superfeedr/documentation/issues/155

Not sure why it fails on certificate error but probably due to being out of date. Either way, without Superfeedr the APK Mirror stuff would need to mass-poll feeds which was trouble in the past.

**Comment by julien51** [comment]
Hi @guzba The certificate on https://simplecast.superfeedr.com/ is up to date... would you be able to tell me more?
Thanks!
